### PR TITLE
Update Solidity compiler to latest version

### PR DESCRIPTION
**Pull request by dbrock**
###### Things done
- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/solc/default.nix [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
   src = fetchFromGitHub {
     owner = "ethereum";
     repo = "solidity";
     rev = "v${version}";
-    sha256 = "1cynqwy8wr63l3l4wv9z6shhcy6lq0q8pbsh3nav0dg9qgj9sg57";
+    sha256 = "1d5x3psz8a9z9jnm30aspfvrpd9kblr14cn5vyl21p27x2vdlzr4";
   };
 
+  patchPhase = ''
+    echo >commit_hash.txt af6afb0415761b53721f89c7f65064807f41cbd3
```
Comment (NeQuissimus) [inline pkgs/development/compilers/solc/default.nix]: a) Is this file really needed?
b) Can you use `pkgs.writeTextFile` instead?

**Comment by NeQuissimus** [comment]
Other than the above, the build LGTM

File: pkgs/development/compilers/solc/default.nix (hunk 1 above)
Comment (dbrock) [inline pkgs/development/compilers/solc/default.nix]: OK, thanks. Still want to use `writeText` @NeQuissimus?

Comment (NeQuissimus) [inline pkgs/development/compilers/solc/default.nix]: No worries
